### ISSUE TITLE
CA1508 incorrect for VB static local var

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn-analyzers#1920]
#### Analyzer package

Microsoft.CodeQuality.Analyzers.Exp

#### Package Version

v2.6.3-beta1.18611.3

#### Diagnostic ID

CA1508 - Remove or refactor conditions to avoid dead code

#### Repro steps

This code incorrectly reports `CA1508: 'Foo Is Nothing' is always 'True'`

```vb.net
Static Foo As String = Nothing
If Foo Is Nothing Then
'  ~~~~~~~~~~~~~~
    Foo = "Foo"
End If
```

#### Expected behavior

No diagnostic.

#### Actual behavior

Diagnostic.
